### PR TITLE
Add support for comma separated srcset values in sef.php

### PR DESCRIPTION
**Pull request by ryandemmer**
System - SEF does not support comma separated values when processing the srcset attribute so only the first URL is converted.

### Summary of Changes

This PR removes srcset from the $attributes array on line 120 and processes instances of the attribute separately, splitting the value by a comma and processing each part individually, then returning the value.

For example,

```html
<img src="images/responsive/winter.jpg" srcset="images/responsive/winter-hd.jpg 2x,images/responsive/winter-2x.jpg 1.5x,images/responsive/winter.jpg 1x" alt="winter" />
```
becomes
```html
<img src="/images/responsive/winter.jpg" srcset="/images/responsive/winter-hd.jpg 2x,/images/responsive/winter-2x.jpg 1.5x,/images/responsive/winter.jpg 1x" alt="winter" /> 
```
whereas before, only the first url in the srcset attribute would have been converted, ie:
```html
<img src="/images/responsive/winter.jpg" srcset="/images/responsive/winter-hd.jpg 2x,images/responsive/winter-2x.jpg 1.5x,images/responsive/winter.jpg 1x" alt="winter" />
```
A more elegant solution may be available, but this one appears to work OK.

### Testing Instructions

Create a new article using the following HTML code, or similar. You will need 3 images of different resolutions, or just three different images of any resolution.
 
```html
<img src="images/image1.jpg" srcset="images/image3.jpg 2x,images/image2.jpg 1.5x,images/image1.jpg 1x" alt="" />
```

### Expected result

The image displays correctly in the browser as all the srcset values are converted.

### Actual result

With the current sef.php the images will not display correctly in a modern browser. You can check the code using the browser console to see that the srcset values have not all been converted.

### Documentation Changes Required

None

### CODE REVIEW

File: plugins/system/sef/sef.php [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * @package     Joomla.Plugin
  * @subpackage  System.sef
  *
- * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
+ * @copyright   Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
```
Comment (brianteeman) [inline plugins/system/sef/sef.php]: This should not be changed :)

**Review by brianteeman — COMMENTED** [review]
Looks like you are working on an old branch which is why the copyright change and it's not showing a real diff

**Comment by zero-24** [comment]
@ryandemmer please take a look into drone ;) http://213.160.72.75/joomla/joomla-cms/2741 Thanks!

**Comment by ryandemmer** [comment]
@zero-24 I have no idea what it is on about.

**Comment by Bakual** [comment]
````
FILE: /drone/src/github.com/joomla/joomla-cms/plugins/system/sef/sef.php
--------------------------------------------------------------------------------
FOUND 2 ERROR(S) AFFECTING 2 LINE(S)
--------------------------------------------------------------------------------
 135 | ERROR | Opening parenthesis of a multi-line function call must be the
     |       | last content on the line
 142 | ERROR | Closing parenthesis of a multi-line function call must be on a
     |       | line by itself
````
You will have to rewrite this lines
````
$buffer = preg_replace_callback($regex, function ($match) use ($base, $protocols) {
	$data = array();
	foreach (explode(",", $match[1]) as $url)
	{
		$data[] = preg_replace('#(?!/|' . $protocols . '|\#|\')([^\s]+)\s+(.*)#', $base . '$1 $2', $url);
	}
	return ' srcset="' . implode(",", $data) . '"';
}, $buffer);
````
to something like
````
$buffer = preg_replace_callback(
	$regex,
	function ($match) use ($base, $protocols)
	{
		$data = array();
		foreach (explode(",", $match[1]) as $url)
		{
			$data[] = preg_replace('#(?!/|' . $protocols . '|\#|\')([^\s]+)\s+(.*)#', $base . '$1 $2', $url);
		}
		return ' srcset="' . implode(",", $data) . '"';
	},
	$buffer
);
````

**Comment by ryandemmer** [comment]
@Bakual Done, but it's still not happy.

Style over substance as far as I'm concerned.

**Comment by Bakual** [comment]
> Done, but it's still not happy.

Codestyle is fine now. I've restarted drone as it had a hickup with JavaScript something. Now it looks fine.

> Style over substance as far as I'm concerned.

To be fair, those lines weren't really easy to read. Not it's much clearer which parts belong together.
So the codestyle rule made sense here.

**Comment by david-bettondesign** [comment]
I have tested this item :white_check_mark: successfully on 1049e451143ff7bb432076080ede545f60af0894<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15300">issues.joomla.org/tracker/joomla-cms/15300</a>.</sub>

**Comment by webfeuerflo** [comment]
I have tested the edited sef.php-file and the srcset-images are working now!

**Comment by ghost** [comment]
@webfeuerflo can you please mark your Test at [issue Tracker](https://issues.joomla.org/tracker/joomla-cms/15300)

**Comment by webfeuerflo** [comment]
I am new to github and not sure how to do that!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15300">issues.joomla.org/tracker/joomla-cms/15300</a>.</sub>

**Comment by ghost** [comment]
- Please open [Issue Tracker](https://issues.joomla.org/tracker/joomla-cms/15300)
- Login with your github-Account
- Click on blue "Test this"-Button above Authors-Picture
- mark your Test as successfully
- hit "submit test result"

**Comment by webfeuerflo** [comment]
I have tested this item :white_check_mark: successfully on 1049e451143ff7bb432076080ede545f60af0894<br /><br />With new sef.php the srcset-images are displayed correctly. did not experience any other issues.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15300">issues.joomla.org/tracker/joomla-cms/15300</a>.</sub>

**Comment by webfeuerflo** [comment]
thank you :-)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15300">issues.joomla.org/tracker/joomla-cms/15300</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by ryandemmer** [comment]
Thank you!

**Comment by wronax** [comment]
This modification causes doubled relative root path in the front of each image url in case image url already contains the relative root path. Please see #17919.

**Comment by ryandemmer** [comment]
The foreach loop could be modified to check the URL for the root path, eg:

```php
foreach (explode(",", $match[1]) as $url)
{
	if (strpos($url, $base) === false) {
		$url = preg_replace('#(?!/|' . $protocols . '|\#|\')([^\s]+)\s+(.*)#', $base . '$1 $2', $url);
	}
	$data[] = $url;
}
```

**Comment by wronax** [comment]
Could you create PR with this fix ?

**Comment by wronax** [comment]
Perfect!

**Pull request merged**